url.bs

```diff
@@ -2476,39 +2476,82 @@ background information. [[!HTML]]
+ e.g., <code>https://examplecorp.com@attacker.example/</code>).
+
+ <li><p>Browsers may render a URL without its <a for=url>scheme</a> if the display surface only ever
+ permits a single scheme (such as a browser feature that omits <code>https://</code> because it is
+ only enabled for secure origins). Otherwise, the scheme may be replaced or supplemented with a
+ human-readable string (e.g., "Not secure"), a security indicator icon, or both.
+
+ <li><p>As described in <a>URL serializer</a>, browsers should not serialize null ports.
 </ul>
 
-<p>For the purposes of bidirectional text it should be rendered as if it were in a
-left-to-right embedding. [[!BIDI]]
+<h4 id=url-rendering-elision>Elision</h4>
+
+<p>In a space-constrained display, URLs should be elided carefully to avoid misleading the user when
+making a security decision:
+
+<ul>
+ <li><p>Browsers should ensure that at least the <a for=host>registrable domain</a> can be shown
+ when the URL is rendered (to avoid showing, e.g., <code>...examplecorp.com</code> when loading
+ <code>https://not-really-examplecorp.com/</code>).
+
+ <li><p>When the full <a for=url>host</a> cannot be rendered, browsers should elide domain labels
+ starting from the lowest-level domain label. For example, <code>examplecorp.com.evil.com</code>
+ should be elided as <code>...com.evil.com</code>, not <code>examplecorp.com...</code>. (Note that
+ bidirectional text means that the lowest-level label may not appear at the left.)
+</ul>
 
-<p class="note no-backref">Unfortunately, as rendered <a for=/>URLs</a> are strings and can appear
-anywhere, a specific bidirectional algorithm for rendered <a for=/>URLs</a> would not see wide
-adoption. Bidirectional text interacts with the parts of a <a for=/>URL</a> in ways that can cause
-the rendering to be different from the model. Users of bidirectional languages are thus cautioned
-that this is to be expected, particularly in plain text environments.
+<h4 id=url-rendering-i18n>Internationalization and special characters</h4>
 
-<p>Due to the confusion that can arise between a <a for=/>URL</a>'s <a for=url>host</a>
-and <a for=url>path</a> with bidirectional text, browsers are encouraged to only render a
-<a for=/>URL</a>'s <a for=url>host</a> in places where it is important for users to
-distinguish between the two. E.g., users are expected to make trust decisions based on a
-<a for=/>URL</a>'s <a for=url>host</a> rendered in the address bar.
+<p>International domain names (IDNs), special characters, and bidirectional text should be handled
+with care to prevent spoofing:
 
+<ul>
+ <li><p>Browsers should render a <a for=/>URL</a>'s <a for=url>host</a> using
+ <a>domain to Unicode</a>.
 
+ <p class="note no-backref">Note that non-ASCII characters can be used in <a
+ href="http://unicode.org/faq/idn.html#26">homograph</a> spoofing attacks. Consider detecting <a
+ href="http://www.unicode.org/reports/tr39/#Confusable_Detection">confusable characters</a> or
```

Comment:
Should this be "and" rather than "or"? Just 'detecting' confusables does not imply anything about how that would affect the rendering of the URL. Chrome shows punycode in some of the cases where confusables show up, which arguably is also not really a "warning". Maybe "... and altering the URL rendering to avoid user confusion." ?

(To be clear, I don't want perfect to be the enemy of good here, or overly focus on the IDN issues, so just noting this problem and a loosely phrased suggestion seems fine, certainly as a first step.)